test/modern_bpf/event_class/event_class.cpp

```diff
@@ -0,0 +1,272 @@
+#include "event_class.h"
+#include <time.h>
+
+/* Messages. */
+#define VALUE_NOT_CORRECT ">>>>> value of the param is not correct. Param id = "
+#define VALUE_NOT_ZERO ">>>>> value of the param must not be zero. Param id = "
+
+/////////////////////////////////
+// CONFIGURATION
+/////////////////////////////////
+
+event_test::event_test(ppm_event_type event_type)
+{
+	m_event_type = event_type;
+	m_current_param = 0;
+
+	/*
+	 * Cleaning phase.
+	 */
+
+	/* 1 - disable the capture of all events. */
+	pman_disable_capture();
+
+	/* 2 - clean all the ring_buffers until they are empty. */
+	clear_ring_buffers();
+
+	/* 3 - clean all interesting syscalls. */
+	mark_all_64bit_syscalls_as_uninteresting();
+
+	/* 4 - detach all bpf programs attached to the kernel a part from syscall dispatchers. */
+	/* Right now we don't have BPF programs to detach here ...*/
+}
+
+void event_test::mark_single_64bit_syscall_as_interesting(int interesting_syscall_id)
+{
+	pman_mark_single_64bit_syscall_as_interesting(interesting_syscall_id);
+}
+
+void event_test::mark_all_64bit_syscalls_as_uninteresting()
+{
+	pman_mark_all_64bit_syscalls_as_uninteresting();
+}
+
+void event_test::enable_capture()
+{
+	pman_enable_capture();
+}
+
+void event_test::disable_capture()
+{
+	pman_disable_capture();
+}
+
+void event_test::clear_ring_buffers()
+{
+	int consume_ret = 1;
+	uint16_t cpu_id = 0;
+	void* event = NULL;
+	while(consume_ret != -1)
+	{
+		consume_ret = pman_consume_one_from_buffers(&event, &cpu_id);
+	}
+}
+
+void event_test::parse_event()
```

Comment:
Can you use `scap_event_decode_params` which is how we usually do parameter decoding? Check the `load_params()` function in libsinsp for an usage example.

Reply:
Hi @LucaGuerra! This is a good point! The main problem that I see here is that this framework test doesn't include the `scap` library, it seems reasonable in my opinion since we are testing just the output of our drivers, so a lower level... The second minor point is that the logic is not exactly the same:
* We don't want to take the minimum between `event_info->nparams` and `evt->nparams` we want always the `evt->nparams` since we want to test it. There could be strange cases in which we take the number in the table without noticing it and this could lead to errors in the testing process
*  The logic is similar but not exactly the same, as you can notice here in `event_test::parse_event()` we fill a first empty parameter in the vector in order to allow users to use humane indexes starting from `1`